src/structlog/formatters.py

```diff
@@ -0,0 +1,39 @@
+# This file is dual licensed under the terms of the Apache License, Version
+# 2.0, and the MIT License.  See the LICENSE file in the root of this
+# repository for complete details.
+
+"""
+Processor->Formatter proxy.
+Formatter for native ``logging`` module that uses bound ``structlog`` processor
+to format message.
+"""
+
+import logging
+
+
+class ProcessorFormatter(logging.Formatter):
+    """Custom stdlib logging formatter for structlog ``event_dict`` messages.
+
+    Apply a structlog processor to the ``event_dict`` passed as
+    ``LogRecord.msg`` to convert it to loggable format (a string).
+
+    """
+
+    def __init__(self, processor, *args, **kwargs):
+        """Keep reference to the ``processor``."""
+        super(ProcessorFormatter, self).__init__(*args, **kwargs)
+        self.processor = processor
+
+    def format(self, record):
+        """Extract structlog's ``event_dict`` from ``record.msg``.
+
+        Process a copy of ``record.msg`` since the some processors modify the
+        ``event_dict`` and the ``LogRecord`` will be used for multiple
+        formatting runs.
+
+        """
+        if isinstance(record.msg, dict):
+            msg_repr = self.processor(
+                record._logger, record._name, record.msg.copy())
+            return msg_repr
+        return record.getMessage()
```

Comment:
I would suggest that you change the implementation to call the `super` formatter as well
``` python
        if isinstance(record.msg, dict):
            msg_repr = self.processor(
                record._logger, record._name, record.msg.copy())
            record.msg = msg_repr
        formatted_msg = super(ProcessorFormatter, self).format(record)
        return formatted_msg
```

that way we can configure additional formatting in the logging config e.g.
``` python
            'plain': {
                '()': ProcessorFormatter,
                'processor': structlog.dev.ConsoleRenderer(colors=False),
               'format': '\n%(message)s [in %(pathname)s:%(lineno)d]'
            },
```